components/brave_rewards/resources/rewards_panel/lib/extension_host.ts

```diff
@@ -0,0 +1,461 @@
+          chrome.braveRewards.tipSite(tabInfo.id, publisherInfo.id, 'one-time')
+          closePanel()
+        }
+      }).catch(console.error)
+    },
+
+    handleMonthlyTipAction (action) {
+      getCurrentTabInfo().then((tabInfo) => {
+        if (!tabInfo) {
+          return
+        }
+
+        const { publisherInfo } = stateManager.getState()
+        if (!publisherInfo) {
+          return
+        }
+
+        const tabId = tabInfo.id
+        const publisherId = publisherInfo.id
+
+        switch (action) {
+          case 'update':
+            chrome.braveRewards.tipSite(tabId, publisherId, 'set-monthly')
+            break
+          case 'cancel':
+            chrome.braveRewards.tipSite(tabId, publisherId, 'clear-monthly')
+            break
+        }
+
+        closePanel()
+
+      }).catch(console.error)
+    },
+
+    handleExternalWalletAction,
+
+    handleNotificationAction (action) {
+      switch (action.type) {
+        case 'open-link':
+          openTab((action as OpenLinkAction).url)
+          break
+        case 'backup-wallet':
+          openTab('chrome://rewards#manage-wallet')
+          break
+        case 'claim-grant':
+          loadCaptcha((action as ClaimGrantAction).grantId, 'pending')
+          break
+        case 'add-funds':
+          handleExternalWalletAction('add-funds')
+          break
+        case 'reconnect-external-wallet':
+          handleExternalWalletAction('reconnect')
+          break
+      }
+    },
+
+    dismissNotification (notification) {
+      chrome.rewardsNotifications.deleteNotification(notification.id)
+      const { notifications } = stateManager.getState()
+      stateManager.update({
+        notifications: notifications.filter(n => n.id !== notification.id)
+      })
+    },
+
+    solveGrantCaptcha (solution) {
+      const { grantCaptchaInfo } = stateManager.getState()
+      if (!grantCaptchaInfo) {
+        return
+      }
+
+      function mapResult (result: number): GrantCaptchaStatus {
+        switch (result) {
+          case 0: return 'passed'
+          case 6: return 'failed'
+          default: return 'error'
+        }
+      }
+
+      const json = JSON.stringify({
+        captchaId: grantCaptchaInfo.id,
+        x: Math.round(solution.x),
+        y: Math.round(solution.y)
+      })
+
+      const grantId = grantCaptchaInfo.grantInfo.id
+
+      chrome.braveRewards.attestPromotion(grantId, json, (result) => {
+        const { grantCaptchaInfo } = stateManager.getState()
+        if (!grantCaptchaInfo || grantCaptchaInfo.grantInfo.id !== grantId) {
+          return
+        }
+
+        const status = mapResult(result)
+
+        stateManager.update({
+          grantCaptchaInfo: { ...grantCaptchaInfo, status }
+        })
+
+        // If the user failed the captcha, request a new captcha for the user.
+        if (status === 'failed') {
+          loadCaptcha(grantId, 'failed')
+        }
+      })
+    },
+
+    clearGrantCaptcha
```

Comment:
Any reason to skip parens here?

Reply:
This is just shorthand for:

```ts
clearGrantCaptcha: clearGrantCaptcha
```

where `clearGrantCaptcha` is a function above.

Reply:
Ah right, totally misread what this was doing - thanks!